src/System.Text.Json/src/System/Text/Json/ThrowHelper.Serialization.cs

```diff
@@ -189,16 +172,26 @@ public static void AddExceptionInformation(in ReadStack readStack, in Utf8JsonRe
             string path = readStack.JsonPath();
             ex.Path = path;
 
-            // If the message is empty, use a default message with Path, LineNumber and BytePositionInLine.
-            if (string.IsNullOrEmpty(ex.Message))
+            string message = ex.Message;
+
+            // If the message is empty or contains EmbedPathInfoFlag then append the Path, LineNumber and BytePositionInLine.
+            if (string.IsNullOrEmpty(message))
             {
+                // Use a default message.
                 Type propertyType = readStack.Current.JsonPropertyInfo?.RuntimePropertyType;
                 if (propertyType == null)
                 {
                     propertyType = readStack.Current.JsonClassInfo.Type;
                 }
 
-                ex.SetMessage($"{SR.Format(SR.DeserializeUnableToConvertValue, propertyType)} Path: {path} | LineNumber: {lineNumber} | BytePositionInLine: {bytePositionInLine}.");
+                message = $"{SR.Format(SR.DeserializeUnableToConvertValue, propertyType)}";
```

Comment:
Why does this need to use string-interpolation?

Reply:
It was a typo from the original; changed. Thanks